shunting-yard-common/src/main/java/com/hotels/shunting/yard/common/event/apiary/SerializableApiaryAlterPartitionEvent.java

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (C) 2016-2018 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.shunting.yard.common.event.apiary;
+
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.hadoop.hive.metastore.api.FieldSchema;
+import org.apache.hadoop.hive.metastore.api.Partition;
+import org.apache.hadoop.hive.metastore.events.AlterPartitionEvent;
+
+import com.hotels.shunting.yard.common.event.SerializableListenerEvent;
+
+public class SerializableApiaryAlterPartitionEvent extends SerializableListenerEvent {
+  private static final long serialVersionUID = 1L;
+
+  private String protocolVersion;
+  private String dbName;
+  private String tableName;
+  private Map<String, String> partitionKeys;
+  private List<String> partitionValues;
+  private List<String> oldPartitionValues;
+
+  SerializableApiaryAlterPartitionEvent() {}
+
+  public SerializableApiaryAlterPartitionEvent(AlterPartitionEvent event) {
+    super(event);
+    dbName = event.getTable().getDbName();
+    tableName = event.getTable().getTableName();
+    partitionKeys = new LinkedHashMap<>();
+    Partition partition = event.getNewPartition();
+    for (FieldSchema fieldSchema : partition.getSd().getCols()) {
+      partitionKeys.put(fieldSchema.getName(), fieldSchema.getType());
+    }
+    partitionValues = partition.getValues();
+
+    oldPartitionValues = event.getOldPartition().getValues();
+  }
+
+  public String getProtocolVersion() {
+    return protocolVersion;
```

Comment:
This would always be null, right? Same with the other SerializableApiary* classes that have `getProtocolVersion()`?

Reply:
No. Its the version of the JSON message coming from Apiary. It won't be null. But I can't get this value from the Hive Events.

Reply:
So how will it be initialised?

Reply:
So, this gets initialized by Jackson because the JSON event coming from Apiary does have a protocol version. I am not using it in SY anywhere but need to have this variable here to map the json properly.